CMakeLists.txt

```diff
@@ -151,19 +154,33 @@ endfunction()
 
 #respect CMAKE_SYSTEM_VERSION
 if(WIN32)
-    if(${CMAKE_SYSTEM_VERSION} EQUAL 10)
-        add_definitions(-D_WIN32_WINNT=0x0A00)
-    elseif(${CMAKE_SYSTEM_VERSION} EQUAL 6.3)
-        add_definitions(-D_WIN32_WINNT=0x0603)
-    elseif(${CMAKE_SYSTEM_VERSION} EQUAL 6.2)
-        add_definitions(-D_WIN32_WINNT=0x0602)
-    elseif(${CMAKE_SYSTEM_VERSION} EQUAL 6.1)
-        add_definitions(-D_WIN32_WINNT=0x0601)
-    elseif(${CMAKE_SYSTEM_VERSION} EQUAL 6.0)
-        add_definitions(-D_WIN32_WINNT=0x0600)
-    else()
-        add_definitions(-D_WIN32_WINNT=0x0501)
-    endif()
+	if(${CMAKE_SYSTEM_VERSION} MATCHES "^10")
+		add_definitions(-D_WIN32_WINNT=0x0A00)
+		set(SYSTEM_LIBRARIES "pathcch")
```

Comment:
don’t we need both `pathcch` and `shlwapi`?

Other than that, please move this whole block down below the `add_library` definition, so you can actually use the `target_X` set of functions to not set the definitions globally.

Reply:
moved and changed to `target_`